packages/app/src/composables/useTransactionData.ts

```diff
@@ -72,30 +72,45 @@ export default (context = useContext()) => {
     try {
       isDecodePending.value = true;
       decodingError.value = "";
+
+      // Get contract ABI first to properly handle verification errors
       await getABICollection([transactionData.contractAddress]);
-      const abi = ABICollection.value[transactionData.contractAddress];
-      if (!abi) {
-        if (!isABIRequestFailed.value) {
-          throw "contract_not_verified";
-        }
-        throw "contract_request_failed";
+
+      // Check for request failures first
+      if (isABIRequestFailed.value) {
+        throw new Error("contract_request_failed");
       }
-      let method = abi ? decodeDataWithABI(transactionData, abi) : undefined;
-      if (!method) {
-        const proxyInfo = await getContractProxyInfo(transactionData.contractAddress);
-        method = proxyInfo?.implementation.verificationInfo
-          ? decodeDataWithABI(transactionData, proxyInfo.implementation.verificationInfo.artifacts.abi)
-          : undefined;
+
+      const contractAbi = ABICollection.value[transactionData.contractAddress];
+      const proxyInfo = await getContractProxyInfo(transactionData.contractAddress);
```

Comment:
`getContractProxyInfo` performs multiple calls under the hood. Can we call it only when `contractAbi` doesn't work?